e2e/files.spec.ts

```diff
@@ -99,7 +99,62 @@ const saveFileSharingSettings = async (page: Page) => {
     page.getByRole('heading', { name: 'File Sharing Settings' }),
   ).not.toBeVisible()
 
-  await confirmAndClosePermissionsSavedAlert(page)
+  await dismissAlert(page, 'Permissions were successfully saved to Synapse')
+}
+
+const uploadFile = async (
+  page: Page,
+  filePath: string,
+  uploadType: 'initialUpload' | 'newVersion',
+) => {
+  await testAuth.step('open file upload modal', async () => {
+    const uploadButtonText =
+      uploadType === 'initialUpload'
+        ? 'Upload or Link to a File'
+        : 'Upload a New Version of File'
+    await page.getByRole('button', { name: uploadButtonText }).click()
+  })
+
+  await testAuth.step('choose file', async () => {
+    const fileChooserPromise = page.waitForEvent('filechooser')
+    await page.getByRole('button', { name: 'Browse...' }).click()
+    if (uploadType === 'initialUpload') {
+      await page
+        .getByRole('menu')
+        .getByRole('link')
+        .filter({ hasText: 'Files' })
+        .click()
+    }
+    const fileChooser = await fileChooserPromise
+    await fileChooser.setFiles(path.join(__dirname, filePath))
+  })
+
+  await testAuth.step('wait for file upload modal to close', async () => {
+    await expect(page.getByText('Initializing......')).not.toBeVisible()
+    await expect(
+      page.getByRole('heading', { name: 'Upload or Link to File' }),
+    ).not.toBeVisible()
+  })
+
+  await testAuth.step('ensure there was not an upload error', async () => {
+    await expect(
+      page.getByRole('heading', { name: 'Upload Error' }),
+    ).not.toBeVisible()
+  })
```

Comment:
Useful for distinguishing between a file upload timeout vs a file upload error when debugging a failed test in CI.